torch/csrc/jit/custom_graph_executor_impl.cpp

```diff
@@ -0,0 +1,61 @@
+#include <torch/csrc/jit/custom_graph_executor_impl.h>
+
+namespace torch {
+namespace jit {
+
+extern const Symbol kDefaultExecutor =
+    Symbol::fromQualString("executor::default");
+
+namespace {
+
+struct GraphExecutorImplState {
+  Symbol name = kDefaultExecutor;
+  bool impl_cached = false;
+  GraphExecutorImplCreator impl_fn;
+};
+thread_local GraphExecutorImplState executor_state;
+
+/* Meyers Singleton */
+std::unordered_map<Symbol, GraphExecutorImplCreator>& getGraphExecutorImpls() {
+  static std::unordered_map<Symbol, GraphExecutorImplCreator> executors;
+  return executors;
+}
+
+} // namespace
+
+Symbol getGraphExecutorName() {
+  return executor_state.name;
+}
+
+void setGraphExecutorName(Symbol name) {
```

Comment:
nitpick: I'm not sure if we need to cache a function pointer. We probably won't have too many executors, so I don't know if we should be worried about optimizing a hash map access. just my 2 cents.